File: _150_evaluate_reverse_polish_notation/main.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stack;
        
        for (int i = 0; i < tokens.size(); i++) {
            // Token is a number
            if (tokens[i].size() > 1 || isdigit(tokens[i][0])) {
                stack.push(stoi(tokens[i]));
            } else {
                int b = stack.top();
                stack.pop();
                int a = stack.top();
                stack.pop();

                switch (tokens[i][0]) {
                    case '+':
                        stack.push(a + b);
                        break;
                    case '-':
                        stack.push(a - b);
                        break;
                    case '*':
                        stack.push(a * b);
                        break;
                    case '/':
                        stack.push(a / b);
                        break;
                }
            }
        }

        return stack.top();
    }
};

File: _150_evaluate_reverse_polish_notation/main.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in range(len(tokens)):
            if len(tokens[i]) > 1 or tokens[i][0].isdigit():
                stack.append(int(tokens[i]))
            else:
                b = stack.pop()
                a = stack.pop()
                
                if tokens[i][0] == '+':
                    stack.append(a + b)
                elif tokens[i][0] == '-':
                    stack.append(a - b)
                elif tokens[i][0] == '*':
                    stack.append(a * b)
                elif tokens[i][0] == '/':
                    stack.append(int(a / b))

        return stack.pop()
